stl/inc/cmath

```diff
@@ -1335,6 +1335,31 @@ _NODISCARD auto hypot(const _Ty1 _Dx, const _Ty2 _Dy, const _Ty3 _Dz) {
 }
 
 #if _HAS_CXX20
+template <class _Ty>
+_NODISCARD constexpr _Ty _Linear_for_lerp(const _Ty _ArgA, const _Ty _ArgB, const _Ty _ArgT) {
+    if (_STD is_constant_evaluated()) {
+        auto _Smaller     = _ArgT;
+        auto _Larger      = _ArgB - _ArgA;
+        auto _Abs_smaller = _Float_abs(_Smaller);
+        auto _Abs_larger  = _Float_abs(_Larger);
+        if (_Abs_larger < _Abs_smaller) {
+            _STD swap(_Smaller, _Larger);
+            _STD swap(_Abs_smaller, _Abs_larger);
+        }
+
+        // _Larger is too large to be subnormal, so scaling by 0.5 is exact, and the product _Smaller * _Larger is large
+        // enough that if _ArgA is subnormal, it will be too small to contribute anyway and this way can sometimes avoid
+        // overflow problems
+        if (_Abs_smaller > 1) {
+            return 2 * (_Ty{0.5} * _ArgA + _Smaller * (_Ty{0.5} * _Larger));

```

Comment:
This is a great explanation, thanks! One last suggestion - this is describing something that is only true when the `if` branch is taken. Therefore, the comment should be within the `if` branch.

(If the comment were phrased conditionally, then it would be fine to place before the condition.)